### ISSUE TITLE
neither Quick Start nor other ways with neo4j >= 4 are working

### ISSUE THREAD

**Issue by derbytau** [neo4j-contrib/neo4j-helm#41]
When using the Quick Start Guide like this:

`helm install mygraph https://github.com/neo4j-contrib/neo4j-helm/releases/download/4.1.0-2/neo4j-4.1.0-2.tgz --set core.standalone=true --set acceptLicenseAgreement=yes --set neo4jPassword=mySecretPassword`

the container is starting but remains in CrashLoopBackoff with the following Error Message:

> Configuration override prefix = mygraph_neo4j_core_0
> Starting Neo4j CORE 0 on mygraph-neo4j-core-0.mygraph-neo4j.mw-internal.svc.cluster.local
> Changed password for user 'neo4j'.
> APOC couln't set a URLStreamHandlerFactory since some other tool already did this (e.g. tomcat). This means you cannot use s3:// or hdfs:// style URLs in APOC. This is caused by a limitation of the JVM which we cannot fix. 
> Fetching versions.json for Plugin 'apoc' from https://neo4j-contrib.github.io/neo4j-apoc-procedures/versions.json
> Installing Plugin 'apoc' from https://github.com/neo4j-contrib/neo4j-apoc-procedures/releases/download/4.1.0.0/apoc-4.1.0.0-all.jar to /plugins/apoc.jar 
> Applying default values for plugin apoc to neo4j.conf
> /var/lib/neo4j/bin/neo4j: line 183: /var/lib/neo4j/conf/neo4j.conf: Permission denied

disabling apoc results only in the last part of the error log:

> Changed password for user 'neo4j'.
> /var/lib/neo4j/bin/neo4j: line 183: /var/lib/neo4j/conf/neo4j.conf: Permission denied

The only way I could fix this isssue was by changing the ImageTag like this:
`helm install mygraph https://github.com/neo4j-contrib/neo4j-helm/releases/download/4.1.0-2/neo4j-4.1.0-2.tgz --set imageTag="3.4.5" --set core.standalone=true --set acceptLicenseAgreement=yes --set neo4jPassword=mySecretPassword`

I think the repository is supposed to support neo4j >= 4, so the 3.4.5 Image should not work but the neo4j 4 versions should work and the neo4j 3 Version already worked with the older helm chart.
It seems like the initial neo4j.conf changes are done  by another user with more permissions than the user that starts the database

**Comment by moxious** [comment]
@derbytau have you previously deployed a different cluster or instance with the name "mygraph"?   The permission denied issues you're running into suggest that you maybe already had a persistent volume claim under the same name that was bound, and perhaps neo4j is running into trouble stepping on data that already existed.

I tried exactly your deploy command at the top and it worked, so I can't reproduce this.   2 specific follow-up steps to try:

* `helm uninstall mygraph` and then manually delete the PVC that is left over afterwards, then reinstall
* Try installing under a different name you've never used before (guaranteeing there's no leftover PVC) and see if the problem still occurs.

**Comment by moxious** [comment]
@derbytau I haven't heard back and so I'm closing this issue.  The resolution above is what I think would fix the issue.  But if you try it and this resolution fails, feel free to ping back, we will re-open and continue to investigate.

**Comment by derbytau** [comment]
sorry for the late answer.
Thanks for your fast answer.
After testing your solutions that didn't work for us, we still searched for the reason this was happening.
We found out that after the root user updated the neo4j.conf file in the startup script of the docker image
the user could not open the neo4f.conf file anymore.
Even if the user was the owner of the file.

We tested in a separate environment the reason behind this issue.
It turned out that there is a bug with the combination of the current `sed` version and the kernels of the OS the kubernetes Nodes are working on.

This result was that when the root user changed a file with a sed command the userrights and ownership of the file stayed the same but the user could not access the file without another change of the userrights.

I hope noone will ever have the same issue but if so this was our reason for the issue.
So I want to document this here and the only solutions (we know of) where 

-  updating the kernel
-  after applying the `sed` do another `chown` or `chmod`

Nevertheless thank you @moxious  for your support and your quick answer
